### PR TITLE
json: Build GStreamer with VAAPI plugin

### PR DESCRIPTION
**Pull request by EliasOfWaffle**

### CODE REVIEW

**Comment by EliasOfWaffle** [comment]
image

**Comment by EliasOfWaffle** [comment]
for enable i think is necessary to user define the hardware-acceleration-policy 'always'

**Comment by EliasOfWaffle** [comment]
Strange is not decoding using vaapi, i think i had make a error with a Youtube open in amount of tabs in my firefox,

**Comment by A6GibKm** [comment]
You are missing https://github.com/flathub/com.belmoussaoui.Decoder/blob/master/com.belmoussaoui.Decoder.json#L70.

**Comment by EliasOfWaffle** [comment]
no are this, is the codec kkkkk 

> You are missing https://github.com/flathub/com.belmoussaoui.Decoder/blob/master/com.belmoussaoui.Decoder.json#L70.

oh thanks i will try now

File: org.gnome.Epiphany.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
         "--socket=wayland",
         "--system-talk-name=org.freedesktop.GeoClue2"
     ],
-    "modules": [
+   "modules": [
```
Comment (A6GibKm) [inline org.gnome.Epiphany.json:17]: This change is not needed.

File: org.gnome.Epiphany.json [hunk 2]
```diff
@@ -144,6 +144,22 @@
                 }
             ]
         },
+        {
+           "name": "gstreamer",
+             "cleanup": [
+               "*"
+          ],
+           "config-opts": [
+              "-Dvaapi=enabled"
```
Comment (A6GibKm) [inline org.gnome.Epiphany.json:153]: Here are the possible flags https://gitlab.freedesktop.org/gstreamer/gstreamer/-/blob/main/subprojects/gstreamer-vaapi/meson_options.txt?ref_type=heads.

File: org.gnome.Epiphany.json [hunk 3]
```diff
@@ -144,6 +144,22 @@
...
+            ],
+            "buildsystem": "meson",
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://gitlab.freedesktop.org/gstreamer/gstreamer",
+                    "commit": "1.22.6"
+                }]
```
Comment (A6GibKm) [inline org.gnome.Epiphany.json:161]: the square bracker `]` should go in the next line.

**Comment by mcatanzaro** [comment]
> for enable i think is necessary to user define the hardware-acceleration-policy 'always'

No, that's for page rendering, not video decoding.

I'm going to close this:

 * [The vaapi element is denylisted and won't be used no matter what](https://github.com/flathub/org.gnome.Epiphany/pull/62#issuecomment-1739136683). Did you miss my comment there? It's disabled for a reason: it's too broken.
 * We don't want to bundle GStreamer. It's a complex and security-sensitive dependency. The GStreamer in the runtime should be good enough.
 * Hardware decoding should use the va element, not the vaapi element. It's [already enabled](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/435c131e5f643d04d01e430e2ade5aa254cb816e/elements/components/gstreamer-plugins-bad.bst#L54) so we don't need to build GStreamer at all.
 * Even the va element has no chance to work so long as the web process has no access to video decoders. Next step is to fix https://github.com/flatpak/flatpak/issues/5524.

**Comment by EliasOfWaffle** [comment]
> * e va eleme

oh 

> > for enable i think is necessary to user define the hardware-acceleration-policy 'always'
> 
> No, that's for page rendering, not video decoding.
> 
> I'm going to close this:
> 
>     * [The vaapi element is denylisted and won't be used no matter what](https://github.com/flathub/org.gnome.Epiphany/pull/62#issuecomment-1739136683). Did you miss my comment there? It's disabled for a reason: it's too broken.
> 
>     * We don't want to bundle GStreamer. It's a complex and security-sensitive dependency. The GStreamer in the runtime should be good enough.
> 
>     * Hardware decoding should use the va element, not the vaapi element. It's [already enabled](https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/blob/435c131e5f643d04d01e430e2ade5aa254cb816e/elements/components/gstreamer-plugins-bad.bst#L54) so we don't need to build GStreamer at all.
> 
>     * Even the va element has no chance to work so long as the web process has no access to video decoders. Next step is to fix [[Feature request]: Sandbox Exemption for V4L2 Hardware Encoders/Decoders flatpak/flatpak#5524](https://github.com/flatpak/flatpak/issues/5524).

oh thanks, sorry for my mistake.
i have a doubt, epiphany have support for ffmpeg? if is possible is not more easy to just use ffmpeg-full extension? or it's is not possible, sorry for my question

**Comment by mcatanzaro** [comment]
> i have a doubt, epiphany have support for ffmpeg?

Via gstreamer-libav, yes. It cannot use ffmpeg directly.

> if is possible is not more easy to just use ffmpeg-full extension? or it's is not possible, sorry for my question

I don't want to add this extension to the manifest because this would enable use of encumbered codes that we don't have in Fedora. I only test one configuration and don't want flatpak stable users to have an entirely different multimedia experience than Tech Preview and Fedora users. It makes it hard for developers to notice and fix bugs.

It's also not needed if your goal is to make hardware decoding work. What's needed for that is to fix https://github.com/flatpak/flatpak/issues/5524.

**Pull request closed without merging**